File: src/main/java/com/psl/PenisStarLeague/dto/LeagueDTO.java
package com.psl.PenisStarLeague.dto;

import java.util.Set;

import com.psl.PenisStarLeague.model.Game;

public record LeagueDTO(int idLeague,
        String league,
        String owner,
        String isPrivate,
        int memberCount,
        String description,
        String leaguePositionCode, 
        Set<UserLeagueDTO> users,
        Set<PendingUserLeagueDTO> pendingUsers,
        Set<Game> games,
        Set<LeagueEventDTO> events) {

}

File: PenisStarLeague-ng/src/app/dto/LeagueDTO.ts
import { Game } from "../model/Game";
import { LeagueEventDTO } from "./LeagueEventDTO";

export class LeagueDTO {
    public idLeague: number | undefined;
    public league: string | undefined;
    public owner: number | undefined;
    public isPrivate: string | undefined;
    public leaguePositionCode: string | undefined; 
    public memberCount: string | undefined;
    public description: string | undefined
    public users: any[] = []; 
    public pendingUsers: any[] = []; 
    public games: Game[] = [];
    public events: LeagueEventDTO[] = [] 
}